forumMessages/views.py

```diff
@@ -4,15 +4,20 @@
 
 
 def ViewMessages(request):
+    print("here")
     messages = Message.objects.order_by('-date')
     if request.method == "POST":
         form = MessageForm(request.POST)
         if form.is_valid():
-            form.save()
-            return redirect(ViewMessages)
+            msg = Message.objects.create(
```

Comment:
what happens if form is not valid?

Reply:
It's used to trigger cleaning on the fields by Django (input validation).
[See ref here:] (https://docs.djangoproject.com/en/3.1/ref/forms/validation/)
Theoretically, if not valid, nothing will be saved and the user will be redirected nonetheless (`redirect `happens after the if's). Though I don't think it's a possible case for a user to get there, thanks to the input validation.

Reply:
makes sense